etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/TransformationRuleEngine.java

```diff
@@ -0,0 +1,81 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.RuleEngine;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.RuleLoader;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.RuleLoaderException;
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import gov.hhs.cdc.trustedintermediary.wrappers.formatter.TypeReference;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+
+/** Implements the RuleEngine interface. It represents a rule engine for transformations. */
+public class TransformationRuleEngine implements RuleEngine {
+    private String ruleDefinitionsFileName;
+    final List<TransformationRule> rules = new ArrayList<>();
+
+    private static final TransformationRuleEngine INSTANCE = new TransformationRuleEngine();
+
+    @Inject Logger logger;
+    @Inject RuleLoader ruleLoader;
+
+    public static TransformationRuleEngine getInstance(String ruleDefinitionsFileName) {
+        INSTANCE.ruleDefinitionsFileName = ruleDefinitionsFileName;
```

Comment:
This will run into concurrency issues. We probably need to synchronize how we initialize the instance. My question now is how is this going to work when we get to the point of each client loading their settings into the engine. Right now, this is set up to take one file with all the clients' settings. In the future, each client will have a settings file. This is a singleton class, the instance is shared across all API instances that are running. Should each API instance have a different instance of the engine?

Reply:
This might work for the first instance if there is only one thread and one settings file. The problem might be more apparent when we start to load individual settings files for each client. I'm picturing a scenario where two threads are loading different settings files to the singleton instance. Do we want to address this now or do it when each client have their own settings file?